Develop

thedayse opened the pull request:
Changes proposed in this pull request:

- remove phpjs library
- update jquery ui and add i18n datapicker
- set a global javascript var with the users date format used to format and validate dates in the timesheet entry
- use the duration to set the endtime if the enddate is blank

Reason for this pull request:
- Because nobody where I work will use this software if the dates are d.m.Y We need Y-m-d

simonschaufi commented:
Thank you for your first pull request! Could you please cleanup your code of any commented out code which is not helpful and any console.log?
please also make sure you follow our code style ;) we don't have a javascript checker but it should look nice and clean.

thedayse commented:
I am new to git so I have probably done everything wrong. The edits are mostly so that users can enter dates in in their own preferred format. Y-m-d is what I prefer. It turned out to be quite complicated because another dd.mm.yyyy is hardcoded all through the code and what look to be outdated zend date functions and the the phpjs library are used. Jquery ui datepicker can format dates so I updated that with the i18n parts that belong to it and added it to git. I've tested it locally and it seems to work.

thedayse commented:
I'll remove all the commented out stuff and console log stuff and submit it again. Not today though. Code style - yes I'll try to get that right as well.

simonschaufi commented:
congratulations for making a pull request in the right branch! many others have failed that test already ;) Usually you wouldn't make changes in your own dev branch but create a new branch based on origin/develop because if you want to do another pull request in develop, you first have to clean up your develop branch. So after this thing is done, better delete your local dev branch.

simonschaufi commented on installer/processor.php:
```diff
@@ -78,10 +78,10 @@ function getpass()
      */
     case "checkRequirements":
-        if (version_compare(PHP_VERSION, '5.4') < 0) {
+		if (PHP_VERSION_ID < 50400){
```
why did you change that?

thedayse replied:
On my ubuntu server with php7 the version test failed. I had to fix it to get it to install..

simonschaufi replied:
I have Ubuntu 16.04.3 LTS with php 7.1 and it works.

simonschaufi replied:
the PHP_VERSION constant prints our `7.1.12-3+ubuntu16.04.1+deb.sury.org+1`

thedayse replied:
mine prints 
7.1.11-0ubuntu0.17.10.1 
I only remember that I had to change it. The version id is always a number and a better test though.

simonschaufi commented on includes/func.php:
```diff
@@ -257,7 +257,9 @@ function checkDBversion($path)
 {
-    $explode_in = explode("-", $in);
+    
+	echo $in, $out;
```
should be removed

simonschaufi commented on installer/steps/25_system_requirements.php:
```diff
@@ -36,8 +36,8 @@
     <h2>System Requirements</h2>
-    The following conditions must be met:<br/>
-    <div class="sp_phpversion fail">at least PHP version 5.4</div>
+    The following conditions must be met:  <?php //echo PHP_VERSION_ID ?> <br/>
```
this is not related to any date picker changes

thedayse replied:
It wouldn't install on php7..

simonschaufi replied:
it will. I have it running on php 7.1

thedayse replied:
I agree - version compare shouldn't have prevented my install. Something did and the version was red and next wasn't clickable, so I started looking at the version check code. I'll revert this

simonschaufi commented on updater/updater.php:
```diff
@@ -1182,6 +1182,22 @@
+);";
+    exec_query($query);
+  
+}
```
please put that in a new pull request

simonschaufi commented on templates/scripts/core/main.php:
```diff
@@ -50,6 +65,8 @@
     <!-- /Extension JavaScripts -->
 
     <script type="text/javascript">
+		
```
remove those empty lines

simonschaufi commented on templates/scripts/core/main.php:
```diff
@@ -1,5 +1,8 @@
 <!DOCTYPE html>
-<html>
+<html lang="<?php //echo '';
+// this breaks the phpjs/strftime library that is out of date anyway - strftime does not have swedish
```
why this comment?

thedayse replied:
I was about to remove that..  setting a unsupported locale tag on the html element caused a javascript exception if phpjs was included. So I removed it. I the html is ever to pass wai/wacg then it need to set language/locale

thedayse commented:
I think it's ready now..

kevinpapst commented on core/processor.php:
```diff
@@ -119,17 +119,13 @@
             die();
         }
-
+		Kimai_Logger::logfile("timeframe " . $axValue);
```
Is it possible that you are formatting with tabs? Please switch to spaces for Kimai.
This seems to be the case in almost all places, I am not going to comment on everyone, but only some examples.

thedayse replied:
yep I'm a tabman. I will try to become a spaceman

simonschaufi replied:
use a proper IDE like phpstorm and tabs will become spaces automatically it setup correctly ;)

kevinpapst commented on extensions/ki_timesheets/js/ts_func.js:
```diff
@@ -252,7 +248,7 @@ function ts_ext_reload_activities(project, noUpdateRate, activity, timeSheetEntr
 /**
  * reloads budget
- * 
+ *
```
Don't take that personal, but the more unneccessary changes you make, the more complicated the review will be and the more comments you will receive ;-)

kevinpapst commented on updater/updater.php:
```diff
@@ -1182,6 +1182,9 @@
     exec_query("ALTER TABLE `${p}configuration` CHANGE `option` `option` VARCHAR(190) NULL");
 }
 
+
```
revert these empty line changes

kevinpapst commented on templates/scripts/core/main.php:
```diff
@@ -42,61 +54,51 @@
         var revision = <?php echo $this->kga['revision'] ?>;
         var timeframeDateFormat = "<?php echo $this->escape($this->kga->getDateFormat(2)) ?>";
-
+		var dateFormat = '<?php echo $this->escape($this->kga->getDateFormat(0)) ?>';
```
tabs vs spaces

kevinpapst commented on templates/scripts/core/main.php:
```diff
@@ -17,20 +17,32 @@
+    <script src="../libraries/jQuery/jquery.newsticker.pack.js"></script>
+    <script src="../libraries/jQuery/js.cookie-2.1.0.min.js"></script>
+    <script src="../libraries/jQuery/jquery-ui.min.js"></script>
+	 <!-- conditional locale include - todo use same for timepicker -->
```
there is a todo left, is that comment still valid?

kevinpapst commented on language/sv.php:
```diff
@@ -235,7 +239,16 @@
     'weekdays_short' => array("Sön", "Mon", "Tis", "Ons", "Tor", "Fri", "Lör"),
-
+	
+	'quicklink_today' => 'Idag',
```
weird formatting

kevinpapst commented on extensions/ki_timesheets/templates/scripts/floaters/add_edit_timeSheetEntry.php:
```diff
@@ -388,31 +393,13 @@ function (data) {
-                    }
-
-                    if (inVal > outVal) {
+				if (startDate.getTime() > endDate.getTime()) {
```
formatting

kevinpapst commented on extensions/ki_timesheets/processor.php:
```diff
@@ -569,31 +569,32 @@ function($activity) {
+				/*  use duration rather than endtime - this only handles minutes */
+				Kimai_Logger::logfile("using minutes duration instead of endtime: ".$minutes);
+				$validoutDate = DateTime::createFromFormat($kga->getDateFormat(3).'  H:i:s', $_REQUEST['start_day'].' '.$_REQUEST['start_time'])->modify("+$minutes minutes");
+				// could do it this way: $validoutDate->add(new DateInterval("PT".$minutes."M"));
```
?

kevinpapst commented on extensions/ki_timesheets/processor.php:
```diff
@@ -569,31 +569,32 @@ function($activity) {
-            }
-        }
-
+     	Kimai_Logger::logfile("outdate: ".$_REQUEST['end_day'] .' '. $_REQUEST['end_time'].', ' .$kga->getDateFormat(3).' H:i:s');
```
do we need all these log calls?

thedayse commented:
I will break this up into smaller chunks and come back with a new PR presently.. Thanks for your help..
